### PR TITLE
[ spi_host, doc ] Update documentation to include deltas from recent PRs

### PR DESCRIPTION
**Pull request by martin-lueker**
- Removes all discussion of the core clock domain
- Eliminates all discussion of CDC's
  - One exception remains the description on "Back-to-Back Commands".
    This section will be updated in a forthcoming PR.
- New class of error event associated with various "impossible" TX FIFO write
  events (i.e., zero-byte writes, or other writes not associated with any known
  processor data type.)
- Generally cleaned up discussion of error events.
- Documents the split of the DATA memory window into separate TXDATA/RXDATA windows
- Implementation section: documents the split of the FSM `cmd_end` signal
  into `last_read` and `last_write` signals

Signed-off-by: Martin Lueker-Boden <martin.lueker-boden@wdc.com>

### CODE REVIEW

**Review by rasmus-madsen — APPROVED** [review]
LGTM

**Review by rasmus-madsen — APPROVED** [review]
LGTM

File: hw/ip/spi_host/doc/_index.md [hunk 1]
```diff
@@ -549,20 +542,23 @@ All data in the waveform is transferred using 32-bit instructions.
...
 
 Filtering out disabled bytes consumes clock cycles in the data pipeline, and can create bubbles in the transmission of SPI_DATA.
 In the worst case, such bubbles can also be interpreted as transient underflow conditions in the TX FIFO, and could trigger spurious interrupts.
 The longest delays occur whenever a word is loaded into the TX FIFO with only one byte enabled.
 
-Zero-byte writes to the {{< regref "DATA" >}} window are not expected.
-Should such transactions ever occur in verification, they will trigger a block level assertion.
+When writing to the {{< regref "TXDATA" >}} window, only three types of data are expected: individual bytes, half-words, and full-words.
+Other types of write transactions (i.e., non-contiguous, zero-byte and three-byte writes) are not supported by most processors.
+Therefore if it is assumed that if such transactions do appear, it is likely a sign of a system integrity error, and thus these other classes of writes are not supported.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:556]:
```suggestion
Therefore it is assumed that if such transactions do appear, it is likely a sign of a system integrity error, and so these other classes of writes are not supported.
```

File: hw/ip/spi_host/doc/_index.md [hunk 2]
```diff
@@ -549,20 +542,23 @@ All data in the waveform is transferred using 32-bit instructions.
...
+
+Should such transactions ever occur, they will trigger an "Invalid Access" error event, which will suspend the processing of future commands until the error has been cleared by setting the {{< regref "ERROR_STATUS.ACCESSINVAL" >}} bit.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:558]:
```suggestion
If such transactions ever occur, they trigger an "Invalid Access" error event, which suspends the processing of future commands until the error has been cleared by setting the {{< regref "ERROR_STATUS.ACCESSINVAL" >}} bit.
```

File: hw/ip/spi_host/doc/_index.md [hunk 3]
```diff
@@ -549,20 +542,23 @@ All data in the waveform is transferred using 32-bit instructions.
...
 
 The RX FIFO has no special provisions for packing received data in any unaligned fashion.
-Depending on the ByteOrder parameter, the first byte received is always packed into either the most- or least-significant byte read from the {{< regref "DATA" >}} memory window.
+Depending on the ByteOrder parameter, the first byte received is always packed into either the most- or least-significant byte read from the {{< regref "RXDATA" >}} memory window.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:561]:
```suggestion
Depending on the `ByteOrder` parameter, the first byte received is always packed into either the most- or least-significant byte read from the {{< regref "RXDATA" >}} memory window.
```

File: hw/ip/spi_host/doc/_index.md [hunk 4]
```diff
@@ -615,30 +611,37 @@ The exact conditions for these *transient* stall conditions are implementation d
...
+- The IP asserts {{< regref "ERROR_STATUS.UNDERFLOW" >}} if it software attempts to read {{< regref "RXDATA" >}} when the RX FIFO is empty.
+- Specifying a command segment with an invalid width (speed), or making a request for a Bidirectional Dual- or Quad-width segement will trigger a {{< regref "ERROR_STATUS.CMDINVAL" >}} error event.
+- Submitting a command segment to an invalid CSID (one larger or equal to `NumCS`) will trigger a {{< regref "ERROR_STATUS.CSIDINVAL" >}} event.
+- {{< regref "ERROR_STATUS.ACCESSINVAL" >}} is asserted if the IP receives a write event to the {{< regref "TXDATA" >}} window that does not correspond to any known processor data type (byte, half- or full-word).
 
-By default all of these programming violations will cause an `error` interrupt when they occur.
+All of these programming violations will create an error event when they occur.
 They will also halt the IP until the corresponding bit is cleared in the {{< regref "ERROR_STATUS" >}} register.
+When every any error event occurs, the error must be acknowledged by clearing (write 1 to clear) the corresponding bit in {{< regref "ERROR_STATUS" >}}.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:624]:
```suggestion
Whenever an error event occurs, the error must be acknowledged by clearing (write 1 to clear) the corresponding bit in {{< regref "ERROR_STATUS" >}}.
```

File: hw/ip/spi_host/doc/_index.md [hunk 5]
```diff
@@ -615,30 +611,37 @@ The exact conditions for these *transient* stall conditions are implementation d
...
+
+By default all error events will trigger an `error` interrupt.
+Clearing the bit corresponding bit in the {{< regref "ERROR_ENABLE" >}} register in the will suppress interrupts for that class of error event and will allow the IP to proceed even if one of these errors has occurred.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:627]:
```suggestion
Clearing the bit corresponding bit in the {{< regref "ERROR_ENABLE" >}} register in the suppresses interrupts for that class of error event and allows the IP to proceed even if one of these errors has occurred.
```

File: hw/ip/spi_host/doc/_index.md [hunk 6]
```diff
@@ -615,30 +611,37 @@ The exact conditions for these *transient* stall conditions are implementation d
...
 
 ## Status Indicators
 
 The {{< regref "STATUS" >}} register contains a number of fields that should be queried for successful operation or troubleshooting.
 
 The register {{< regref "STATUS.ACTIVE" >}} indicates whether a command segment is currently being processed by the FSM.
-Even if {{< regref "STATUS.ACTIVE" >}} is high it is often still possible to insert another command segment into the CDC, allowing for the possibility of back-to-back processing of multiple segments.
-So the register {{< regref "STATUS.READY" >}} indicates that there is room to submit another segment into {{< regref "COMMAND" >}}.
+Even if {{< regref "STATUS.ACTIVE" >}} is high it is often still possible to insert another command segment into the command FIFO.
+So the register {{< regref "STATUS.READY" >}} indicates that there is room in the command FIFO.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:642]:
```suggestion
The register {{< regref "STATUS.READY" >}} indicates that there is room in the command FIFO.
```

File: hw/ip/spi_host/doc/_index.md [hunk 7]
```diff
@@ -653,7 +656,7 @@ Before any commands are processed, the block must be enabled by writing one to t
 Writing a zero to this register temporarily suspends any previously submitted transactions.
 If the block is re-enabled by writing a one to {{< regref "CONTROL.SPIEN" >}}, any previously executing commands will continue from wherever they left off.
 
-An unacknowledged error interrupt will also suspend the core state machine.
+An unacknowledged error event will also suspend the core state machine.
```
Comment (martin-lueker) [inline hw/ip/spi_host/doc/_index.md:659]:
```suggestion
An unacknowledged error event suspends the core state machine.
```

**Review by martin-lueker — COMMENTED** [review]
Online edits

**Pull request merged**